include/rtc.h

```diff
@@ -134,116 +93,121 @@ extern "C" {
- * do need to do this, make sure to call #rtc_close BEFORE #timer_close
- * and then call #rtc_init again after you call #timer_init to restart
- * the Timer Subsystem!
+ * @brief Close the RTC subsystem, disabling system hooks.
  */
 void rtc_close( void );
 
 /**
- * @brief Determine whether the RTC supports writing the time.
+ * @brief Check if the specified RTC source is usable by the subsystem.
+ */
+bool rtc_is_source_available( rtc_source_t source );
+
+/**
+ * @brief Get the current source clock for the subsystem.
+ */
+rtc_source_t rtc_get_source( void );
+
+/**
+ * @brief Switch the preferred source clock for the subsystem.
  *
- * Some emulators and flash carts do not support writing to the RTC, so
- * this function makes an attempt to detect silent write failures and will
- * return `false` if it is unable to change the time on the RTC.
+ * By default, the subsytem will use to the first available source,
+ * but some games may wish to specify the preferred RTC source.
  *
- * This function is useful if your program wants to conditionally offer the
- * ability to set the time based on hardware/emulator support.
+ * Make sure you call #rtc_resync_time after switching sources!
  *
- * Unfortunately this operation may introduce a slight drift in the clock,
- * but it is the only way to determine if the RTC supports the write command.
+ * @return whether the new source clock was successfully selected
+ */
+bool rtc_set_source( rtc_source_t source );
+
+/**
+ * @brief Resynchronize the subsystem's time with the source clock.
  *
- * This operation will take approximately 1 second to complete.
+ * You should only need to do this after switching sources.
  *
- * @return whether RTC writes appear to be supported
+ * @return whether the source clock was successfully synchronized
  */
-bool rtc_is_writable( void );
+bool rtc_resync_time( void );
 
 /**
  * @brief Read the current date/time from the real-time clock.
  *
- * If the RTC is not detected or supported, this function will
- * not modify the destination rtc_time parameter.
- *
- * Your code should call this once per frame to update the #rtc_time_t
- * data structure. The RTC Subsystem maintains a cache of the
- * most-recent RTC time that was read and will only perform an
- * actual RTC read command if the cache is invalidated. The
- * destination rtc_time parameter will be updated regardless of
- * the cache validity.
- *
- * Cache will invalidate every #RTC_GET_CACHE_INVALIDATE_TICKS.
- * Calling #rtc_set will also invalidate the cache.
- *
- * If an actual RTC read command is needed, this function can take
- * a few milliseconds to complete.
- *
- * @param[out]  rtc_time
- *              Destination pointer for the RTC time data structure
- *
- * @return whether the rtc_time destination pointer data was modified
+ * @return the current RTC time as a UNIX timestamp
  */
-bool rtc_get( rtc_time_t * rtc_time );
+time_t rtc_get_time( void );
```

Comment:
We might want to move these to some internal header, as they're only useful as internal implementation. For API, time() is sufficient.

Reply:
Done https://github.com/DragonMinded/libdragon/pull/563/commits/9c0b45291f0dee239eb48ae485da21b901d76fe6